src/segmentmytif/features.py

```diff
@@ -25,42 +30,112 @@ def from_string(s):
+def get_features(
+    raster: xr.DataArray,
+    input_path: Path,
+    feature_type: FeatureType,
+    features_path: Path,
+    chunk_overlap: int = 200,
+    mode: Literal["normal", "parallel", "safe"] = "normal",
+    **extractor_kwargs,
+):
+    """Extract features from the input data, or load them from disk if they have already been extracted.
+
     :param input_data: 'Raw' input data as stored in TIFs by a GIS user. Shape: [n_bands, height, width]
     :param input_path:
     :param feature_type: See FeatureType enum for options.
     :param features_path: Path used for caching features
-    :param profile:
+    :param chunk_overlap: Overlap between chunks when chunk-wise processing is enanbled
     :param extractor_kwargs: options for the feature extractor
     :return:
     """
     if feature_type == FeatureType.IDENTITY:
-        return input_data
-    else:
-        if features_path is None:
-            features_path = get_features_path(input_path, feature_type)
-        if not features_path.exists():
-            logger.info(f"No existing {feature_type.name} features found at {features_path} for input data with shape {input_data.shape}")
-            with log_duration(f"Extract {feature_type.name} features", logger):
-                features = extract_features(input_data, feature_type, **extractor_kwargs)
-            log_array(features, logger, array_name=f"{feature_type.name} features")
-            logger.info(f"Saving {feature_type.name} features (shape {features.shape}) to {features_path}")
-            save_tiff(features, features_path, profile)
-        loaded_features, _ = read_geotiff(features_path)
-        logger.info(f"Loading {feature_type.name} features (shape {loaded_features.shape}) from {features_path}")
-        return loaded_features
-
-
-def extract_features(input_data, feature_type, **extractor_kwargs):
+        return raster
+
+    if features_path is None:
+        features_path = get_features_path(input_path, feature_type)
+    if not features_path.exists():
+        # Extract features
+        msg = (
+            f"No existing {feature_type.name} features found at {features_path} "
+            f"for input data with shape {raster.data.shape}"
+        )
+        logger.info(msg)
+
+        with log_duration(f"Extracting {feature_type.name} features", logger):
+            features_data = extract_features(raster.data, feature_type, chunk_overlap, **extractor_kwargs)
+        log_array(features_data, logger, array_name=f"{feature_type.name} features")
+
+        # Create xarray DataArray with the extracted features
+        # Keep the geospatial information from the input raster
+        features = xr.DataArray(
+            features_data,
+            dims=raster.dims,
+            coords={"band": range(features_data.shape[0]), "y": raster.y, "x": raster.x},
+        )
+        features.rio.write_crs(raster.rio.crs, inplace=True)
```

Comment:
Add a issue investigate if everything is copied

Reply:
Make sure not to copy too much. Maybe related: #52

Reply:
Hi @cwmeijer, I could not reproduce this. See details in my reply in #52